tests/coverage.rs

```diff
@@ -0,0 +1,122 @@
+mod logging;
+mod server;
+
+use failure::Error;
+use headless_chrome::{Browser, LaunchOptionsBuilder};
+
+use headless_chrome::browser::tab::Tab;
+use server::Server;
+use std::sync::Arc;
+
+fn basic_http_response(
+    body: &'static str,
+    content_type: &'static str,
+) -> tiny_http::Response<&'static [u8]> {
+    tiny_http::Response::new(
+        200.into(),
+        vec![tiny_http::Header::from_bytes(&b"Content-Type"[..], content_type.as_bytes()).unwrap()],
+        body.as_bytes(),
+        Some(body.len()),
+        None,
+    )
+}
+
+fn server_with_html_and_js() -> Server {
+    Server::new(|request: tiny_http::Request| {
+        let url = request.url();
+
+        let content_type = if url.ends_with(".js") {
+            "application/javascript"
+        } else {
+            "text/html"
+        };
+
+        let body = if url.ends_with("coverage_fixture1.js") {
+            include_str!("coverage_fixtures/coverage_fixture1.js")
+        } else if url.ends_with("coverage_fixture2.js") {
+            include_str!("coverage_fixtures/coverage_fixture2.js")
+        } else {
+            include_str!("coverage_fixtures/basic_page_with_js_scripts.html")
+        };
+
+        let response = basic_http_response(body, content_type);
+        request.respond(response)
+    })
+}
+
+#[test]
+fn returns_actual_coverage() -> Result<(), Error> {
+    logging::enable_logging();
+    let server = server_with_html_and_js();
+    let browser = Browser::new(
+        LaunchOptionsBuilder::default()
+            .headless(true)
+            .build()
+            .unwrap(),
+    )
+    .unwrap();
+    let tab: Arc<Tab> = browser.wait_for_initial_tab()?;
+
+    tab.start_js_coverage()?;
+
+    let url = format!("http://127.0.0.1:{}", server.port());
+    tab.navigate_to(&url)?;
+
+    tab.wait_until_navigated()?;
+
+    let script_coverages = tab.take_precise_js_coverage()?;
+
+    // our fixtures HTML file (basic_page_with_js_scripts.html) includes two external scripts
+    assert_eq!(2, script_coverages.len());
+
+    let first_script = script_coverages
+        .iter()
+        .find(|script_coverage| script_coverage.url.ends_with("coverage_fixture1.js"))
+        .unwrap();
+
+    // in coverage_fixture1.js, there are two anonymous functions and one nameless "function" that
+    // you can imagine as being the global scope (i.e. the top level scope of the script)
+    assert_eq!(3, first_script.functions.len());
+
+    // this one should not have been executed yet:
+
+    let on_click_function_coverage = first_script
+        .functions
+        .iter()
+        .find(|function_coverage| function_coverage.function_name == "button.onclick")
+        .unwrap();
+
+    let fn_range = on_click_function_coverage.ranges.first().unwrap();
+    assert_eq!(fn_range.count, 0);
+
+    let button = tab.wait_for_element("#incrementor")?;
+    button.click()?;
+
+    let updated_script_coverages = tab.take_precise_js_coverage()?;
```

Comment:
There is an implicit counter reset here?

Reply:
I tried to cover that in the comment below this line about only getting "updates", but I'll add a line to the docs of this function.